hw/ip/i2c/doc/_index.md

```diff
@@ -117,6 +117,18 @@ In this state the output drivers `scl_tx_o` and `sda_tx_o` are controlled direct
 When {{< regref OVRD.SCLVAL >}} and {{< regref OVRD.SDAVAL >}} are set high, the virtual open drain configuration will leave the output resistively pulled high, and controllable by remote targets.
 In this state, with SCL or SDA asserted high, the register fields {{< regref VAL.SCL_RX >}} and {{< regref VAL.SDA_RX >}} can be used to receive inputs (including remote acknowledgments) from target devices.
 
+#### FSM control of SCL and SDA
+
+While in host mode, SCL and SDA are generated through the internal state machine.
+Since SCL is directly decoded from the states, it can have short glitches during transition which the external target may be sensitive to if it is not using an over-sampling scheme.
+To counter this, the SCL and SDA outputs from the internal state machine are flopped before they are emitted.
+
+This adds a one cycle module clock delay to both signals.
+If the module clock is sufficiently faster than I2C line speeds (for example 20MHz), this is not an issue.
+However if the line speeds and the module clock speeds become very close (2x), the 1 cycle delay may have an impact, as the internal state machine may mistakenly think it has sampled an SDA that has not yet been updated.
+
+It it thus recommended to run clock ratios such that the internal module clock is at least 5x-10x the line speeds.
```

Comment:
i think it still is, but i haven't looked super closely or tried it out.  For one thing I think the synchronization delay being greater than the line speed / clock speed ratio is already going to cause problems.  Since by the time you actually see the input you're already into a different clock phase.